impexp-core/src/main/java/org/citydb/database/adapter/oracle/WorkspaceManagerAdapter.java

```diff
@@ -74,4 +80,21 @@ public boolean gotoWorkspace(Connection connection, Workspace workspace) {
 		}
 	}
 
+	@Override
+	public List<String> fetchWorkspacesFromDatabase(Connection connection) throws SQLException {
+		try (Statement stmt = connection.createStatement();
+			 ResultSet rs = stmt.executeQuery("select workspace from all_workspaces order by workspace")) {
```

Comment:
Should we only query those workspaces that are not frozen? Maybe also owned by the user (schema)?

Reply:
I am actually not an Oracle expert for this question. Assume that you are user `AAA` but connect to the database using the schema `BBB` instead. And assume that the schema of `BBB` is version-enabled and `BBB` has created some workspaces. Is the user `AAA` allowed to goto these workspaces? I assume the answer is no unless you have been granted sufficient privileges? Can you grant privileges on a workspace?

Regarding frozen workspaces: Well, we have not checked this in previous versions. But you are right, we might have to consider this. What are the criteria to remove a frozen workspace from the list?

Reply:
Thanks, @yaozhihang, for solving an issue with using workspaces in another schema in the 3DCityDB repo (https://github.com/3dcitydb/3dcitydb/commit/1b00b87f583b6de6153ae61b7748579a22c84fc1).

We kept frozen workspaces in the list of workspaces that can be queried on the Database tab. If the workspace is in `NO_ACCESS` mode, a corresponding error message is now shown to the user (168a5efecf181f52e33bbf3f5dc916f490f22ba1).